added support for the ZMQ_LAST_ENDPOINT option in zmq-get-socket-option

Terramorpha opened the pull request:
currently, running `(zmq-get-socket-option sock ZMQ_LAST_ENDPOINT)` throws an error due to the default size `(sizeof size_t)` this commit detects the `ZMQ_LAST_ENDPOINT` option, sets an appropriate buffer size and converts the opt bytevector back to a string.

mothacehe commented on simple-zmq.scm.in:
```diff
@@ -495,7 +495,6 @@ ASCII etc."
       (if (= result -1)
 		  (zmq-get-error errno)
 		  (let ((len (bytevector-uint-ref size 0 (native-endianness) (sizeof size_t))))
-			(display len) (newline)
```
you should squash this with the previous commit

mothacehe commented on simple-zmq.scm.in:
```diff
@@ -480,20 +480,28 @@ ASCII etc."
         (pointer->message! msg-pointer))))
 
 (define (zmq-get-socket-option socket option)
```
the indentation is wrong and there's a mix of spaces and tabulations

mothacehe commented on simple-zmq.scm.in:
```diff
@@ -480,20 +480,28 @@ ASCII etc."
-                         (sizeof size_t)))
+  (let* ((value-size (cond
+                      ((eq? ZMQ_TYPE option) (sizeof int))
+					  ((eq? ZMQ_LAST_ENDPOINT option) 1024)
```
why 1024?

mothacehe commented on simple-zmq.scm.in:
```diff
@@ -516,6 +524,8 @@ ASCII etc."
     (when (= result -1)
       (zmq-get-error errno))))
 
+
```
you can remove those

mothacehe commented on simple-zmq.scm.in:
```diff
@@ -480,20 +480,28 @@ ASCII etc."
+			(display len) (newline)
+			(cond
+			 ((eq? ZMQ_LAST_ENDPOINT option)
+			  (let ((new-bv (make-bytevector (1- len))))
```
I need to review it in more details but this seems quite complex

mothacehe reviewed (changes requested):
Hello,
Thanks for the patch, here are a few comments.
Mathieu

Commit:
add support for the ZMQ_LAST_ENDPOINT in zmq-get-socket-option

fix unnecessary newline

fix indentation

Terramorpha commented:
I made the changes you requested. I reindented the code using spaces, I squashed the commits and I removed the newline.
About your last comment, `zmq-get-socket-option` interprets the contents of the `opt` bytevector as an unsized integer. However, for the specific `ZMQ_LAST_ENDPOINT` option, it should interpret it as a string.
I added a special case for this.
The `1-`s are because the returned size takes into account the C string's null terminator.

mothacehe commented:
I added a few cosmetic changes, as well as the associated unit tests before pushing. Thanks!
